src/Blocks/BlockDirt.h

```diff
@@ -48,13 +49,15 @@ class cBlockDirtHandler :
 		}
 		else if ((a_RelY < cChunkDef::Height - 1))
 		{
+			BLOCKTYPE above = a_Chunk.GetBlock(a_RelX, a_RelY + 1, a_RelZ);
 			NIBBLETYPE light = std::max(a_Chunk.GetBlockLight(a_RelX, a_RelY + 1, a_RelZ), a_Chunk.GetTimeAlteredLight(a_Chunk.GetSkyLight(a_RelX, a_RelY + 1, a_RelZ)));
```

Comment:
Why? The variable isn't used until line 62. I was suggesting that you put it between the two ifs.